docs/rest_api.md

```diff
@@ -1,4 +1,302 @@
+```
+wsk namespace list -v
+```
+```
+REQUEST:
+[GET]	https://openwhisk.ng.bluemix.net/api/v1/namespaces
+Req Headers
+{
+  "Authorization": [
+    "Basic XXXYYYY"
+  ]
+}
+RESPONSE:Got response with code 200
+Resp Headers
+{
+  "Content-Type": [
+    "application/json; charset=UTF-8"
+  ]
+}
+Response body size is 28 bytes
+Response body received:
+["john@example.com_dev"]
+```
+
+As you can see you the printed information provides the properties of the HTTP request, it performs a HTTP method `GET` on the URL `https://openwhisk.ng.bluemix.net/api/v1/namespaces` using a Basic Authorization header `Basic XXXYYYY`. 
+Notice that the authorization value is your base64-encoded OpenWhisk authorization string.
+The response is of content type `application/json`.
+
+## Actions
+
+To create or update an action send a HTTP request with method `PUT` on the the actions collection. For example, to create a `nodejs:6` action with the name `hello` using a single file content use the following:
+```bash
+curl -u $AUTH -d '{"namespace":"_","name":"hello","exec":{"kind":"nodejs:6","code":"function main(params) { return {payload:\"Hello \"+params.name}}"}}' -X PUT -H "Content-Type: application/json" https://openwhisk.ng.bluemix.net/api/v1/namespaces/_/actions/hello?overwrite=true 
+```
+
+To perform a blocking invocation on an action, send a HTTP request with a method `POST` and body containing the input parameter `name` use the following:
+```bash
+curl -u $AUTH https://openwhisk.ng.bluemix.net/api/v1/namespaces/_/actions/hello?blocking=true \
+-X POST -H "Content-Type: application/json" \
+-d '{"name":"John"}'  
+```
+You get the following response:
+```json
+{
+  "duration": 2,
+  "name": "hello",
+  "subject": "john@example.com_dev",
+  "activationId": "c7bb1339cb4f40e3a6ccead6c99f804e",
+  "publish": false,
+  "annotations": [{
+    "key": "limits",
+    "value": {
+      "timeout": 60000,
+      "memory": 256,
+      "logs": 10
+    }
+  }, {
+    "key": "path",
+    "value": "john@example.com_dev/hello"
+  }],
+  "version": "0.0.1",
+  "response": {
+    "result": {
+      "payload": "Hello John"
+    },
+    "success": true,
+    "status": "success"
+  },
+  "end": 1493327653769,
+  "logs": [],
+  "start": 1493327653767,
+  "namespace": "john@example.com_dev"
+}
+```
+If you just want to get the `response.result`, run the command again with the query parameter `result=true`
+```bash
+curl -u $AUTH "https://openwhisk.ng.bluemix.net/api/v1/namespaces/_/actions/hello?blocking=true&result=true" \
+-X POST -H "Content-Type: application/json" \
+-d '{"name":"John"}' 
+```
+You get the following response:
+```json
+{
+  "payload": "hello John"
+}
+```
+
+## Annotations and Web Actions
+
+To create an action as a web action, you need to add an [annotation](annotations.md) of `web-export=true` for web actions. Since web-actions are publicly accessible, you should protect pre-defined parameters (i.e., treat them as final) using the annotation `final=true`.
```

Comment:
I would add here the note about `--web true` from the CLI that does both annotations.